net/strongswan/files/ipsec.init

```diff
@@ -238,11 +248,11 @@ config_remote() {
 		[ "$remote_gateway" = "%any" ] && ipdest="1.1.1.1" || ipdest="$remote_gateway"
 		local_gateway=$(ip -o route get "$ipdest" | awk '/ src / { print gensub(/^.* src ([^ ]*) .*$/, "\\1", "g"); }')
 	}
-
-	secret_xappend -n "$local_gateway "
-	[ -n "$remote_identifier" ] && secret_xappend -n "$remote_identifier " || secret_xappend -n "$remote_gateway "
-
-	secret_xappend ": PSK \"$pre_shared_key\""
+	if [ "$auth_method" = "psk" ]; then
+		secret_xappend -n "$local_gateway "
+		[ -n "$remote_identifier" ] && secret_xappend -n "$remote_identifier " || secret_xappend -n "$remote_gateway "
```

Comment:
That doesn't look right. Aren't there the [] missing around the last two conditions?
Also, a local and remote identifier aren't strictly required for PSKs. They are just necessary for finding the right secret if multiple can be candidates.

A good solution would be the following:
Remote and local identifiers/IPs are optional, but if more than one PSK is detected without an ID or IP, then issue a warning

Reply:
> That doesn't look right. Aren't there the [] missing around the last two conditions?

No, `[ test ] && then-stmt || else-stmt` is a shorthand notation.

> Also, a local and remote identifier aren't strictly required for PSKs. They are just necessary for finding the right secret if multiple can be candidates.

Knowing if there can be multiple matches might require sophisticated logic to detect that in this script...

Reply:
OK.